src/lib/structures/expr.ml

```diff
@@ -311,393 +311,374 @@ module F_Htbl : Hashtbl.S with type key = t =
-      fprintf fmt "(> %a %a)" print a print b
+      Fmt.pf ppf "(> %a %a)" pp a pp b
 
     | Sy.L_neg_built Sy.LT, [a; b] ->
-      fprintf fmt "(>= %a %a)" print a print b
+      Fmt.pf ppf "(>= %a %a)" pp a pp b
 
     | Sy.L_neg_pred, [a] ->
-      fprintf fmt "(not %a)" print a
+      Fmt.pf ppf "(not %a)" pp a
 
     | Sy.L_built (Sy.IsConstr hs), [e] ->
-      fprintf fmt "((_ is %a) %a)" Hstring.print hs print e
+      Fmt.pf ppf "((_ is %a) %a)" Hstring.print hs pp e
 
     | Sy.L_neg_built (Sy.IsConstr hs), [e] ->
-      fprintf fmt "(not ((_ is %a) %a))" Hstring.print hs print e
+      Fmt.pf ppf "(not ((_ is %a) %a))" Hstring.print hs pp e
 
     | (Sy.L_built (Sy.LT | Sy.LE) | Sy.L_neg_built (Sy.LT | Sy.LE)
       | Sy.L_neg_pred | Sy.L_eq | Sy.L_neg_eq
       | Sy.L_built (Sy.IsConstr _)
-      | Sy.L_neg_built (Sy.IsConstr _)) , _ ->
+      | Sy.L_neg_built (Sy.IsConstr _)), _ ->
       assert false
 
-  and print_silent fmt t =
-    let open Format in
+  and pp_operator = Symbols.pp_operator ~format:`Smtlib
+
+  and pp_silent ppf t =
     let { f ; xs ; ty; bind; _ } = t in
     match f, xs with
-    (* Formulas *)
-    | Sy.Form form, xs -> print_formula fmt form xs bind
+    | Sy.Form form, xs -> pp_formula ppf form xs bind
+
+    | Sy.Lit lit, xs -> pp_lit ppf lit xs
 
     | Sy.Let, [] ->
       let x = match bind with B_let x -> x | _ -> assert false in
-      fprintf fmt
-        "(let%a %a =@ %a in@ %a)"
-        (fun fmt x -> if Options.get_verbose () then
-            fprintf fmt
-              " [sko = %a]" print x.let_sko) x
-        Sy.print x.let_v print x.let_e print_silent x.in_e
+      Fmt.pf ppf "(let ((%a %a)) %a)"
+        Symbols.print x.let_v
+        pp x.let_e
+        pp_silent x.in_e
 
-    (* Literals *)
-    | Sy.Lit lit, xs -> print_lit fmt lit xs
+    | Sy.(Op Record), _ ->
+      begin
+        match ty with
+        | Ty.Trecord { Ty.lbs = lbs; record_constr; _ } ->
+          assert (List.compare_lengths xs lbs = 0);
+          Fmt.pf ppf "(%a %a)"
+            Hstring.print record_constr
+            Fmt.(list ~sep:sp pp) xs
 
-    | Sy.Op Sy.Get, [e1; e2] ->
-      if Options.get_output_smtlib () then
-        fprintf fmt "(select %a %a)" print e1 print e2
-      else
-        fprintf fmt "%a[%a]" print e1 print e2
-
-    | Sy.Op Sy.Set, [e1; e2; e3] ->
-      if Options.get_output_smtlib () then
-        fprintf fmt "(store %a %a %a)"
-          print e1
-          print e2
-          print e3
-      else
-        fprintf fmt "%a[%a<-%a]" print e1 print e2 print e3
+        | _ ->
+          (* Exclude by the typechecker. *)
+          assert false
+      end
 
-    | Sy.Op Sy.Concat, [e1; e2] ->
-      fprintf fmt "%a@@%a" print e1 print e2
+    | Sy.Op op, [] -> Fmt.pf ppf "%a" pp_operator op
 
-    | Sy.Op Sy.Extract (i, j), [e] ->
-      fprintf fmt "%a^{%d,%d}" print e i j
+    | Sy.Op op, _ :: _ ->
+      Fmt.pf ppf "(%a %a)" pp_operator op Fmt.(list ~sep:sp pp) xs
```

Comment:
I guess you meant `"@[(%a@ %a)@]" to be consistent with other printers?

Reply:
Actually I meant what I wrote because I don't like closing boxes after the parentheses (it leads to more unreadable `)))))))))`) but that would work as well as long as we don't print everything as a unit.